### PR TITLE
Covered issues mentioned in #498

### PR DESCRIPTION
**Pull request by mateBakos**
### Description
https://github.com/microsoft/QuantumKatas/issues/498
- Refactored Zip and Zip3 functions to Zipped and Zipped3 respectively
- Removed redundant test in [RandomNumberGeneration](https://github.com/microsoft/QuantumKatas/blob/main/tutorials/RandomNumberGeneration/Tests.qs#L150)

### Side-note
Thank you for making "good first issues" like this one, I hope this will be my first real open-source contribution. I hope I did everything right, please let me know if changes are needed.

### CODE REVIEW

**Comment by vivanwin** [comment]
Hi @mateBakos,

Congratulations on your first contribution 🥳!

Your PR looks good. 

Unfortunately, as you can see that the CI/CD did pass. This is because PR #494 is not yet merged, which updates the QDK to the new version which has the new functions that you have introduced. 
We are working to merge the new QDK version, then this PR can also be merged. 

Thank you for your contribution!

**Comment by mateBakos** [comment]
Hi @vivanwin,

Thanks for the kind words and the opportunity! 

I checked the logs of the CD/CI and was wondering how my changes could have affected so many things, I'm glad to hear that is not the case though.

File: SimonsAlgorithm/BooleanMatrix.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public BooleanVector(IEnumerable<long> array)
                 throw new ArgumentException($"Vectors have different sizes [{vector1.Count} != {vector2.Count}]");
             }
 
-            return new BooleanVector(vector1.Zip(vector2, (val1, val2) => val1 ^ val2));
```
Comment (vivanwin) [inline SimonsAlgorithm/BooleanMatrix.cs:50]: This is not a Q# file. Therefore, you do not need to update this `Zip` function to `Zipped`

File: KeyDistribution_BB84/ReferenceImplementation.qs [hunk 2]
```diff
@@ -82,7 +82,7 @@ namespace Quantum.Kata.KeyDistribution {
         // If Alice and Bob used the same basis, they will have the same value of the bit.
         // The shared key consists of those bits.
         mutable key = new Bool[0];
-        for ((a, b, bit) in Zip3(basesAlice, basesBob, measurementsBob)) {
```
Comment (vivanwin) [inline KeyDistribution_BB84/ReferenceImplementation.qs:85]: There is one `(` added to many.

**Review by vivanwin — CHANGES_REQUESTED** [review]
Hi,
I have added two small comments which are breaking the builds.

**Comment by mateBakos** [comment]
Hi, 

Thanks for the detailed comments, I think they should be fixed now. We'll see once the CD/CI finishes.

Please let me know if there is anything else.

**Review by vivanwin — APPROVED** [review]

**Pull request merged**